modules/post/multi/manage/shell_to_meterpreter.rb

```diff
@@ -96,13 +97,25 @@ def run
     else
       # Find the best fit, be specific with uname to avoid matching hostname or something else
       target_info = cmd_exec('uname -ms')
-      if target_info =~ /linux/i && target_info =~ /86/
-        # Handle linux shells that were identified as 'unix'
+      vprint_status("target_info= #{target_info}")
+      vprint_status("Platform: Linux")
+      if target_info =~ /linux/i
         platform = 'linux'
-        payload_name = 'linux/x86/meterpreter/reverse_tcp'
         lplat = [Msf::Platform::Linux]
-        larch = [ARCH_X86]
-        vprint_status("Platform: Linux")
+        if target_info =~ /86/
```

Comment:
You should flip this and the /64/ check, or `Linux x86_64` will match /86/ before reaching /64/.

Reply:
Or put x86 in the else section...